### ISSUE TITLE
CS1061 "Add Using" code fix also changes the name of the method called

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#28096]
**Version Used**: 15.8.0 Preview 3 (27821.2.d15.8)

**Steps to Reproduce**:

1. Download and extract [this minimal repro solution](https://github.com/dotnet/roslyn/files/2133583/ClassLibrary17.zip)
2. Open ClassLibrary17.sln and Class1.cs
3. Set caret on line 14 and press Ctrl+. to activate quick fix menu.

**Expected Behavior**:

An offer to "add `using NuGet.Protocol;`" (which is what I actually get on Line 13.

**Actual Behavior**:

An offer to "GetCoreV3 - `using NuGet.Protocol;`"

The difference is so subtle that I almost missed the bug it introduced. Both lines need the same using added to work. Why would one want to also change the method called? I've *never* seen Ctrl+. actually change the method called before. That seems like a bug.

![image](https://user-images.githubusercontent.com/3548/41857665-e789c3bc-784c-11e8-9032-1cd11637d132.png)

**Comment by AArnott** [comment]
OK, I just realized the docs I was reading are out of date. There is no longer a `GetCoreV2` method defined in the library. So I guess this quick fix is a result of some (new?) fuzzy matching logic, which is honestly pretty cool -- if it weren't so subtle it would be better though, IMO.

**Comment by jmarolf** [comment]
This behavior has been in since at least 15.3 (@CyrusNajmabadi  to confirm).  If we detect that there is a symbol already defined that is within a certain edit-distance we offer to correct it.  With us knowing about symbols defined in nuget packages we will do this if the only symbol we find is in an unreferenced nuget package.

**Comment by AArnott** [comment]
OK. Well as I say, it's a nifty feature. It just caught me off guard.

**Comment by CyrusNajmabadi** [comment]
Indeed.  This is by design.  Note: i think there is a *lot* of open space here for "calling out" things that are unexpected.  Whether that is through text (i.e.  ```(fuzzy match)```), or through iconography (i.e. an exclamation glyph or something), is completely open.  Tagging @sharwell @kuhlenh to track this as this goes into the general bucket of giving people confidence about the selections they're making.

**Comment by CyrusNajmabadi** [comment]
here's an idea:  as an example, we already have a glyph we use if we're going to *add* a nuget package.  We use that glyph to help draw the user's attention tot he fact that this is going to do something "big".

We should consider the same for "fuzzy matching". i.e. we should have a custom glyph that indicates that a name change is going to happen.  So that the list is much clearer that this is more than just an 'add using'.  What that glyph is isn't clear to me.  But i'm sure we could figure something good out.

**Comment by dpoeschl** [comment]
We'd like to do the following:
- Change the title of the fix to something like "Change to 'Goo' and add using Whatever;"
- Use a glyph to indicate it's a spelling-based fix (for this and also regular spell check).

**Comment by CyrusNajmabadi** [comment]
Closing out as not worth making any changes here given only one report it 4 years.
